src/parse/trainer.py

```diff
@@ -146,9 +149,21 @@ def _produce_trainer_stages(data: dict) -> TrainerSteps:
 
     model = get_model(stages[0]['model'] | { "x_dim" : x_dim }, x_dim, y_dim)
     optimizer = get_optimizer(model, stages[0]['optim'])
-    if 'model_weights' in data and 'optim_state' in data:
+
+    resume = 'model_weights' in data and not 'base_model' in stages[0]['model']
+    lora = 'base_model' in stages[0]['model'] and type(model) == MODELS['lora']
+    lora_resume = lora and 'model_weights' in data and any(['lora' in weight for weight in list(data.get('model_weights', {}).keys())])
```

Comment:
this line also worries me, because it has the underlying assumption that the string 'lora' will be in the finetuning weights, which is not a scalable solution for other finetuning methods, but I'm not sure how to solve this problem otherwise

Reply:
not sure what you mean here, as the `model_weights` are gotten from the CL, and the nested structure could only be created by doing the logic I just did, right? unless we change the interaction with the CL or yml files?